### DISCUSSION TITLE
Extract relationships and table data to csv

### DISCUSSION THREAD

**Discussion by 10xape** [TabularEditor/TabularEditor3#1411] [Data Modeling]
Hi

Is there any option to extract all relationships and table data (or at least sample) from a workspace model to a csv file? 

![image](https://github.com/user-attachments/assets/015e6f07-3f52-4e80-ba5f-f7eab0ebb146)

![image](https://github.com/user-attachments/assets/bba04020-5e33-4c0b-b276-5ec611f22601)

Thanks

Matt

**Comment by otykier** [comment]
Do you wish to extract the data in an automated way? If so, you can use the [C# script feature](https://docs.tabulareditor.com/te2/Useful-script-snippets.html#querying-analysis-services). However, please be advised that querying large amounts of data in one go can potentially cause memory issues on both the server and client side. If your table has many rows (say, more than 100.000) you may want to write some code to process it in batches, e.g. using the [`TOPNSKIP`](https://dax.guide/topnskip) DAX function.

We also have an **Export data...** option in the **Query** menu, after you've executed a query. But please note this has the same limitation as mentioned above. It may be useful for just exporting a sample of data as Excel / CSV, but again, if you have *many* rows, this technique will not work do to memory constraints.

![image](https://github.com/user-attachments/assets/a175c9eb-3824-48b1-adcf-f5601ecf5346)

For exporting relationships, could you please clarify what you need? Is it just the list of relationships? In that case, you could use a C# script like this:

```csharp
ExportProperties(Model.Relationships, "FromColumn,ToColumn,IsActive,CrossFilteringBehavior")
    .Replace("\t", ",")
    .Output();
```

**Reply by 10xape** [reply]
yes I just need data samples  and all relationships basic info, like what it realtes to.

**Reply by 10xape** [reply]
what would be the script to export the masures?

**Reply by otykier** [reply]
Just a list of measures? That would be:

```csharp
ExportProperties(Model.AllMeasures).Output();
```
